Configuration/Config.cs

```diff
@@ -159,26 +159,60 @@ public static JToken GetToken(string key)
         }
 
         /// <summary>
-        /// Sets a configuration value. This is really only used to help testing. The key heye can be
-        /// specified as {environment}.key to set a value on a specific environment
+        /// Sets a configuration value. If isEnvironmentContext is set to true (default) key can be
+        /// specified as {environment}.key to set a value on a specific environment. When false,
+        /// key is considered the full path to the meaning object.
         /// </summary>
         /// <param name="key">The key to be set</param>
         /// <param name="value">The new value</param>
-        public static void Set(string key, string value)
+        /// <param name="isEnvironmentContext">Flag indicates whether to understand key as specific environment key</param>
+        public static void Set(string key, string value, bool isEnvironmentContext = true)
         {
-            JToken environment = Settings.Value;
-            while (key.Contains("."))
+            if (isEnvironmentContext)
```

Comment:
Is there a reason to have this flag? I suppose the only reason would be to support keys that have a dot in them. Do we have such a case, and if so, I would recommend against configuration keys that have periods in them due to the ambiguity that can arise.